test/generate-vdm.spec.ts

```diff
@@ -0,0 +1,167 @@
+  const cli = jest.requireActual('cli-ux');
+  return {
+    ...cli,
+    default: {
+      ...cli.default,
+      confirm
+    }
+  };
+});
+
+const warn = jest.fn(message => console.log('MOCKED WARNING: ', message));
+jest.mock('@oclif/command', () => {
+  const command = jest.requireActual('@oclif/command');
+  command.Command.prototype.warn = warn;
+  return command;
+});
+
+import { GeneratorOptions as GeneratorOptionsSDK, generatorOptionsCli as generatorOptionsSDK } from '@sap/cloud-sdk-generator';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import GenerateVdm, { FlagsParsed } from '../src/commands/generate-vdm';
+import * as foo from '../src/utils/generate-vdm-util';
+
+const spyToGeneratorSDK = jest.spyOn(foo, 'toGeneratorSDK');
+
+describe('generate-vdm', () => {
+  const pathForTests = path.resolve(__dirname, __filename.replace(/\./g, '-')).replace('-ts', '');
+
+  beforeAll(() => {
+    const pathForResources = path.resolve(__dirname, 'resources', 'template-generator-vdm');
+    fs.copySync(pathForResources, pathForTests);
+  });
+
+  afterAll(() => {
+    fs.removeSync(pathForTests);
+  });
+
+  afterEach(() => {
+    spyToGeneratorSDK.mockClear();
+  });
+
+  beforeEach(() => {
+    spyToGeneratorSDK.mockClear();
+  });
+
+  it('should generate a vdm.', async () => {
+    await GenerateVdm.run(['-i', 'edmxSource', '-o', 'generatedVdm', '--forceOverwrite', '--projectDir', pathForTests]);
+    const files = fs.readdirSync(path.join(pathForTests, 'generatedVdm', 'business-partner-service'));
+    expect(files.length).toBeGreaterThan(0);
+  }, 60000);
+
+  it('should fail if the mandatory parameters are not there', async () => {
+    try {
+      await GenerateVdm.run([]);
+    } catch (err) {
+      const messageNoFormatting = err.message as string;
+      expect(messageNoFormatting).toContain('-i, --inputDir INPUTDIR');
+    }
+  });
+
+  it('should parse strings and with --no- it should lead to all false.', async () => {
+    const expected = getAllFalse();
+    delete expected.projectDir;
+    try {
+      await GenerateVdm.run([...getInputAllFalse()]);
+    } catch (e) {
+      expect(spyToGeneratorSDK).toHaveReturnedWith(expected);
+    }
+  });
+
+  it('should resolve the projectdir', async () => {
+    const projectDir = '/root';
+    const expected = getDefault(projectDir);
+
+    try {
+      await GenerateVdm.run(['-i', 'input', '-o', 'output', '--projectDir', projectDir]);
+    } catch (e) {
+      expect(spyToGeneratorSDK).toHaveReturnedWith(expected);
+    }
+  });
+
+  it('should pass each boolean flags correctly', async () => {
+    return Object.keys(generatorOptionsSDK).map(async key => {
+      const option = generatorOptionsSDK[key as keyof GeneratorOptionsSDK];
+      const expected = getDefault('/somePrefix');
+      const args = ['-i', 'input', '-o', 'output', '--projectDir', '/somePrefix'];
+
+      if (option !== undefined && option.type === 'boolean') {
```

Comment:
this is a bad comparision. When checking for undefined, always use `typeof option !== 'undefined'`

In this case it might be enough to check `typeof option !== 'boolean'`.